core/chains/evm/logpoller/log_poller.go

```diff
@@ -149,6 +155,7 @@ func NewLogPoller(orm ORM, ec Client, lggr logger.Logger, pollPeriod time.Durati
 		keepFinalizedBlocksDepth: keepFinalizedBlocksDepth,
 		filters:                  make(map[string]Filter),
 		filterDirty:              true, // Always build Filter on first call to cache an empty filter if nothing registered yet.
+		finalityViolated:         &isReorged,
```

Comment:
nit: we don’t need to put `false`, the doc explicitly states that the zero value is `false`.

> The zero value is false.
-- https://pkg.go.dev/sync/atomic@go1.22.1#Bool

That’s also what most devs should expect because it’s consistent with the standard zero-init rules.

Also, I find it confusing to name the variable `isReorged` in one place and assign it to `finalityViolated` later because reorg != finality violation in general.

```suggestion
		finalityViolated:         new(atomic.Bool),
```